### PR TITLE
#42 inital scafold frontend with next, mui and next-auth

### PR DESCRIPTION
**Pull request by dmijatovic**
@ctwhome and @ewan-escience This is quite big MR, but inevitable when scafolding frontend project ;-)

### CODE REVIEW

File: frontend/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+yarn-debug.log*
+yarn-error.log*
+
+# local env files
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+# vercel
```
Comment (ctwhome) [inline frontend/.dockerignore:33]: No need to ignore vercel, or is it something that comes with nextjs anyway?

File: frontend/components/layout/AppHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      return (
+        <MenuItem
+          key={item.path}
+          item={item}
+        />
+      )
+    })
+  }
+
+  function getLoginButton(){
```
Comment (ctwhome) [inline frontend/components/layout/AppHeader.tsx:38]: Creating several components in the same file is not a good patter for reusablility. Shouldn't be in different files?

File: frontend/components/layout/AppHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        backgroundColor: 'background.paper',
+        zIndex: 8
+      }}
+    >
+      <RSDLogo />
+      <Box component="nav"
+        sx={{
+          display: 'flex',
```
Comment (ctwhome) [inline frontend/components/layout/AppHeader.tsx:76]: Why there are some styles rendered here?

File: frontend/components/layout/AppHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        }}>
+        {getMenuItems()}
```
Comment (ctwhome) [inline frontend/components/layout/AppHeader.tsx:78]: This makes you go back and forward into the code, it reminds me of "go to pattern"

File: frontend/components/layout/CardGrid.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  padding:'1rem 0rem',
+  '@media (min-width: 1024px)':{
+    gridTemplateColumns: '1fr 1fr',
+  },
+  '@media (min-width: 1920px)':{
+    gridTemplateColumns: '1fr 1fr 1fr',
+  }
+}))
+
+export default function CardGrid({children}:{children:any}) {
```
Comment (ctwhome) [inline frontend/components/layout/CardGrid.tsx:17]: Type any. Is it possible to block this type via eslint?

File: frontend/components/layout/ContentInTheMiddle.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+const StyledArticle = styled('article')(({theme})=>({
+  position:'fixed',
+  display:'flex',
+  justifyContent:'center',
+  alignItems:'center',
+  minWidth:'100vw',
+  minHeight:'100vh'
+}))
+
+export default function ContentInTheMiddle({children,...props}:{children:any}) {
```
Comment (ctwhome) [inline frontend/components/layout/ContentInTheMiddle.tsx:12]: The naming are very randomly created, content in the middle can be anything, is it difficult to follow?

File: frontend/components/layout/Copyright.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import Typography from '@mui/material/Typography';
+import Link from '@mui/material/Link';
+
+export default function Copyright(props:any) {
+  return (
+    <Typography variant="body2" align="center" {...props}>
```
Comment (ctwhome) [inline frontend/components/layout/Copyright.tsx:6]: Typography should be a css class, why is it a component?

File: frontend/components/layout/Copyright.tsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+      {'Copyright © '}
+      <Link color="inherit" href="/">
+        Research Software Directory
+      </Link>{' '}
```
Comment (ctwhome) [inline frontend/components/layout/Copyright.tsx:10]: Spaces and dots within the html itself. This creates boilerplate and makes hard to follow

File: frontend/components/layout/Copyright.tsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import Typography from '@mui/material/Typography';
+import Link from '@mui/material/Link';
+
+export default function Copyright(props:any) {
+  return (
+    <Typography variant="body2" align="center" {...props}>
+      {'Copyright © '}
```
Comment (ctwhome) [inline frontend/components/layout/Copyright.tsx:7]: With is this a string literal?

File: frontend/components/layout/MenuItem.tsx [hunk 10]
```diff
@@ -0,0 +1,36 @@
+import styled from '@mui/system/styled'
+
+import {MenuItem} from '../../config/menuItems'
```
Comment (ctwhome) [inline frontend/components/layout/MenuItem.tsx:3]: Use root reference paths like ~/ or @/. Is it possible?

File: frontend/components/layout/MenuItem.tsx [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+const StyledLabel = styled('div')(({})=>({
+  minWidth: '7rem',
+  textAlign: 'center'
+}))
+
+export default function MenuItemLink({item}:{item:MenuItem}) {
+  return (
+    <StyledLink
+      href={item.path}
```
Comment (ctwhome) [inline frontend/components/layout/MenuItem.tsx:27]: This is way too much boilerplate for something that can be done with a couple of classes. It creates boilerplate (more code to write, maintain, difficult to read and prone to error)

File: frontend/components/layout/PageTitle.tsx [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  zIndex: 9,
+  '@media (max-width: 768px)':{
+    flexDirection:'column',
+  },
+}))
+
+export default function PageTitle({title,children}:{title:string,children?:any}) {
+  return (
+    <PageTitleStyled>
+      <Typography variant="h1" sx={{flex:1}} color="secondary">
```
Comment (ctwhome) [inline frontend/components/layout/PageTitle.tsx:21]: Very difficult to understand, and again why styles inside an already styled component?

File: frontend/components/layout/RSDLogo.tsx [hunk 13]
```diff
@@ -0,0 +1,51 @@
+import {styled} from "@mui/system"
+import {nextRouterWithLink} from './nextRouterWithLink'
+
+const LogoSvg = styled('svg')({
+  minWidth: '4rem',
+  minHeight: '4rem',
+  margin: '0rem 1rem 0rem 0rem'
```
Comment (ctwhome) [inline frontend/components/layout/RSDLogo.tsx:7]: This margin will create problems when reuse somewhere else. It is not generic. The positioning style should be in the father

File: frontend/components/layout/RSDLogo.tsx [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+        <defs>
+        <clipPath id="clip0_679_64">
+        <rect width="608" height="608" fill="white"/>
+        </clipPath>
+        </defs>
+      </LogoSvg>
+      {/* <LogoText>Reasearch Software Directory</LogoText> */}
+    </LogoContainer>
+  )
+}
```
Comment (ctwhome) [inline frontend/components/layout/RSDLogo.tsx:51]: Not counting the svg code, this is way too much code for just a simple image with a link.

File: frontend/components/login/GitlabIcon.tsx [hunk 15]
```diff
@@ -0,0 +1,15 @@
+import styled from '@mui/system/styled'
+
+const StyledIcon = styled('svg')({
+  width: '1.5rem',
+  height: '1.5rem',
+  marginRight: '1rem'
```
Comment (ctwhome) [inline frontend/components/login/GitlabIcon.tsx:6]: Same here, let the parent container decide the positioning

File: frontend/components/login/LoginOptions.tsx [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+    callbackUrl:"http://localhost:3000/user/profile"
+  })
+}
+
+export default function LoginOptions() {
+  return (
+    <Box
+      component="section"
+      sx={{
+        display: 'grid',
```
Comment (ctwhome) [inline frontend/components/login/LoginOptions.tsx:23]: Styles in components like this makes it very confusing.

**Review by ctwhome — COMMENTED** [review]
I see how next brings a bunch set of utilities to the library. I am against material ui and style components mit creates a lot of spaghetti code in my opinion and confusing. There is absolutely not separation of concerns between html, js, and css.
I suggest to drop mui and use tailwindcss instead, which will remove boilerplate, is framework agnostic.

File: frontend/.dockerignore (hunk 1 above)
Comment (dmijatovic) [reply frontend/.dockerignore:33]: Yes, it is in default ignore file with next so I took it over. I see no harm if it is already included.

File: frontend/components/layout/CardGrid.tsx (hunk 5 above)
Comment (dmijatovic) [reply frontend/components/layout/CardGrid.tsx]: It is not in default linter rule. But I improved type fot the child elements

File: frontend/components/layout/RSDLogo.tsx (hunk 13 above)
Comment (dmijatovic) [reply frontend/components/layout/RSDLogo.tsx]: Well spotted! I removed it.

File: frontend/components/login/GitlabIcon.tsx (hunk 15 above)
Comment (dmijatovic) [reply frontend/components/login/GitlabIcon.tsx:6]: Goog advice, thanks.

**Comment by dmijatovic** [comment]
@ctwhome Can you please use this branch as starting point for frontend
